packages/loader/driver-utils/src/networkUtils.ts

```diff
@@ -9,24 +9,34 @@ import { ITelemetryLoggerExt } from "@fluidframework/telemetry-utils/internal";
 import { OnlineStatus, canRetryOnError, isOnline } from "./network.js";
 
 /**
+ * Logs network failure details to telemetry.
  * @internal
  */
 export function logNetworkFailure(
 	logger: ITelemetryLoggerExt,
 	event: ITelemetryErrorEventExt,
-	error?: any,
-) {
+	error?: unknown,
+): void {
 	const newEvent = { ...event };
 
-	const errorOnlineProp = error?.online;
+	// Extract online property if present
+	const errorOnlineProp = (error as { online?: string })?.online;
 	newEvent.online =
 		typeof errorOnlineProp === "string" ? errorOnlineProp : OnlineStatus[isOnline()];
 
-	if (typeof navigator === "object" && navigator !== null) {
-		const nav = navigator as any;
-		const connection = nav.connection ?? nav.mozConnection ?? nav.webkitConnection;
-		if (connection !== null && typeof connection === "object") {
-			newEvent.connectionType = connection.type;
+	if (typeof navigator === "object") {
```

Comment:
A bit of research suggests that the navigator object would indeed never be `null`, runtimes that don't support it seem to (probably?) leave it `undefined`, which should work fine with the new code... I guess I'm just curious how confident we are about that, to completely remove the `null` checks.